### PR TITLE
fix: wait for ctx cancel cleanup job finished 

### PR DESCRIPTION
**Pull request by levimm**
there're two goroutines doing cleaning up when context is canceled. Add waitgroup to make sure both are handled before exit.

### CODE REVIEW

File: cmd/kube-egress-gateway-cnimanager/cmd/serve.go [hunk 1]
```diff
@@ -92,7 +93,10 @@ func ServiceLauncher(cmd *cobra.Command, args []string) {
 		logger.Error(err, "failed to create cni config manager")
 		os.Exit(1)
 	}
+	wg := sync.WaitGroup{}
```
Comment (MartinForReal) [inline cmd/kube-egress-gateway-cnimanager/cmd/serve.go:96]: I'd prefer https://pkg.go.dev/golang.org/x/sync/errgroup But this is optional.

**Comment by jwtty** [comment]
pls rebase, otherwise, lgtm

File: cmd/kube-egress-gateway-cnimanager/cmd/serve.go (hunk 1 above)
Comment (levimm) [reply cmd/kube-egress-gateway-cnimanager/cmd/serve.go]: thanks, please take another look at the errgroup usage.

Comment (levimm) [reply cmd/kube-egress-gateway-cnimanager/cmd/serve.go]: thanks, please take another look at the errgroup usage.

**Review by MartinForReal — COMMENTED** [review]
LGTM

**Review by jwtty — APPROVED** [review]

**Pull request merged**